File: chapter11/typescript/MacOS.ts
import { OperatingSystem } from "./OperatingSystem";

/**
 * MacOS implements the bridge "implementation" interface OS
 *
 * @Author Bridget
 */
 export class MacOS implements OperatingSystem {

    public displayOSName(): void {
        console.log("macOS");
    }

    public displayOSVersion(): void {
        console.log("13 (Ventura)");
    }

    public displayMemory(): void {
        console.log("32 GB RAM");
    }

    public start(): void {
        console.log("Start Mac Operating System");
    }

    public shutdown(): void {
        console.log("Shutdown Mac Operating System");
    }
}

File: chapter11/java/designpattern/src/main/java/com/designpattern/MacOS.java
package com.designpattern;

/**
 * MacOS implements the bridge "Implementor" interface OperatingSystem
 *
 * @version 1.0.0
 * @Author Bridget
 * @license MIT
 */
public class MacOS implements OperatingSystem {

    @Override
    public void displayOSName() {
        System.out.println("macOS");
    }

    @Override
    public void displayOSVersion() {
        System.out.println("13 (Ventura)");
    }

    @Override
    public void displayMemory() {
        System.out.println("32 GB RAM");
    }

    @Override
    public void start() {
        System.out.println("Start Mac Operating System");
    }

    @Override
    public void shutdown() {
        System.out.println("Shutdown Mac Operating System");
    }
}